LanguageFeatures/Inline-classes/static_analysis_inline_class_A16_t01.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A compile-time error occurs if an inline class constructor
+/// includes a superinitializer
+///
+/// @description Checks that a compile-time error occurs if an inline class
+/// constructor includes a superinitializer
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+inline class V1 {
+  final int id;
+  V1(this.id);
+  V1.x(this.id);
+}
+
+inline class V2 extends V1 {
```

Comment:
I think we should avoid `extends` because that's already a syntax error (so perhaps we won't even start doing the type analysis).

So `implements V1` is better.

We can still test the superinitializer as if `V1` had allowed us to call its constructor in that manner.

However, we should also test that it is rejected if we try to invoke a constructor of `Object` (which could be done by having a plain `super()` superinitializer).

Reply:
Good! I think we need one more change, though. I added a new comment to line 22.